Assets/Scripts/GameAI/AIAgentManager.cs

```diff
@@ -0,0 +1,83 @@
+﻿namespace GameAI
+{
+    using UnityEngine;
+    using HarmonyQuest.Audio;
+    using Navigation;
+    using AIGameObjects;
+    using System.Collections.Generic;
+
+    public class AIAgentManager : MonoBehaviour
+    {
+        private AIGameObject[] aiGameObjects;
+        private List<AIAgent> agents;
+
+        private float pathRefreshTimer = 0.0f;
+
+        // Start is called before the first frame update
+        void Start()
+        {
+            PopulateAgentsList();
+            FmodMusicHandler.instance.AssignFunctionToOnBeatDelegate(AgentsBeatUpdate);
+        }
+
+        public void PopulateAgentsList()
+        {
+            aiGameObjects = FindObjectsOfType<AIGameObject>();
+            agents = new List<AIAgent>();
+            foreach (AIGameObject aiGameObject in aiGameObjects)
+            {
+                agents.Add(new AIAgent(aiGameObject));
```

Comment:
Using a `foreach` lambda here would be sweet, like I was doing in the `CameraController` [here](https://github.com/nshooter3/HarmonyQuest/blob/stage/Assets/Scripts/DynamicCamera/CameraController.cs).
So I think it'd be:
```c#
aiGameObjects.ForEach(aiGameObject => agents.Add(new AIAgent(aiGameObject)));
```

Reply:
Gonna opt out of this one because converting my array to list for this operation to work adds more complexity.